File: gestion-ressources-ng22/src/app/core/models/signaler-dto.ts
import { PersonelDto } from '..\@app/core/models/personel-dto';
import { ProduitDto } from '..\@app/core/models/produit-dto';
export class SignalerDto {

  id: number=0
  etat: string= "";
produitDto:ProduitDto= new ProduitDto();
personelDto:PersonelDto= new PersonelDto();
}

File: backend/src/main/java/com/odji/spring_back_end/dto/SignalerDto.java
package com.odji.spring_back_end.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SignalerDto {

    private Integer id;
    private String etat;

    // Relations sortantes exposées en DTO (pas en entité !)
    private ProduitDto produit;
    private PersonelDto personel;
}
